tools/code-analyzer/src/git.ts

```diff
@@ -53,13 +53,38 @@ export const cloneRepo = async ( repoPath: string ) => {
 	return folderPath;
 };
 
+export const sparseCheckoutRepo = async (
```

Comment:
I wanted a way to do a sparse checkout as minimal as possible for checking out composer.json files. We could use hard links to raw github files, but this is more flexible for future if we want to sparse checkout any repo.